Service completely broken: Authentication fails with ITMO

# 🐛 Service completely broken: Authentication fails with ITMO

## Problem Description
The service is completely non-functional and fails to authenticate with ITMO's OAuth2 system. All calendar export attempts result in HTTP 500 Internal Server Error.

## Error Details
When trying to access the calendar URL, the service crashes with:

```
AssertionError: Keycloak form action regexp match not found
```

**OR**

```
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://id.itmo.ru/auth/realms/itmo/protocol/openid-connect/token
Response: {'error': 'invalid_grant', 'error_description': 'Code not valid'}
```

## Reproduction Steps
1. Set up the service with valid ITMO credentials
2. Try to access any calendar URL (e.g., `http://localhost:35601/calendar/...`)
3. Service returns "Internal Server Error"
4. Logs show authentication failure

## Current Status
- ❌ **Official Docker image also broken**: `docker run --rm -p 35601:35601 iburakov/my-itmo-ru-to-ical` exhibits the same issue
- ❌ **System-wide problem**: This affects ALL users
- ❌ **Service is completely unusable**

## Root Cause Analysis
It appears that ITMO has changed their authentication system. The current implementation expects HTML forms with specific patterns, but these no longer exist or have changed significantly.

## Impact
**CRITICAL**: This issue makes the entire service completely non-functional for all users. No one can export their ITMO schedule using this tool.

## Request
Could you please investigate and fix the authentication issue? The service has been very useful for ITMO students, and we would appreciate if it could be restored to working condition.

---

## 🇷🇺 Описание на русском

**Проблема:** Сервис полностью сломан - не работает авторизация с ИТМО.

**Ошибки:** 
- `AssertionError: Keycloak form action regexp match not found`
- `400 Client Error: invalid_grant, Code not valid`

**Статус:** Сервис полностью неработоспособен для всех пользователей.

**Просьба:** Пожалуйста, исправьте проблему с авторизацией. Сервис очень полезен для студентов ИТМО.

## Environment
- Any Python/Docker environment
- ITMO authentication system (id.itmo.ru)
- Current repository version

--- a/src/auth.py
+++ b/src/auth.py
@@ -29,7 +29,7 @@ def get_code_challenge(code_verifier: str):
     return code_challenge.replace("=", "")  # remove base64 padding
 
 
-_FORM_ACTION_REGEX = re.compile(r'"registrationAction":\s*"(?P<action>https://id\.itmo\.ru/auth/realms/itmo/login-actions/authenticate\?[^"]*)"', re.DOTALL)
+_FORM_ACTION_REGEX = re.compile(rf'"loginAction":\s*"(?P<action>{re.escape(_PROVIDER)}[^"]*)"', re.DOTALL)
 
 
 @AsyncTTL(time_to_live=55 * 60, skip_args=1)

--- a/src/auth.py
+++ b/src/auth.py
@@ -29,7 +29,7 @@ def get_code_challenge(code_verifier: str):
     return code_challenge.replace("=", "")  # remove base64 padding
 
 
-_FORM_ACTION_REGEX = re.compile(r'<form\s+.*?\s+action="(?P<action>.*?)"', re.DOTALL)
+_FORM_ACTION_REGEX = re.compile(r'"registrationAction":\s*"(?P<action>https://id\.itmo\.ru/auth/realms/itmo/login-actions/authenticate\?[^"]*)"', re.DOTALL)
 
 
 @AsyncTTL(time_to_live=55 * 60, skip_args=1)

